include/experimental/__p0009_bits/extents.hpp

```diff
@@ -237,20 +240,24 @@ class extents
 
 
   MDSPAN_TEMPLATE_REQUIRES(
-    class SizeType,
+    class SizeType, size_t N,
     /* requires */ (
       _MDSPAN_TRAIT(is_convertible, SizeType, size_type)
     )
   )
+  MDSPAN_CONDITIONAL_EXPLICIT(N != rank_dynamic())
   MDSPAN_INLINE_FUNCTION
   constexpr
-  extents(std::array<SizeType, rank_dynamic()> const& dyn) noexcept
+  extents(std::array<SizeType, N> const& exts) noexcept
 #if defined(_MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS)
     : __storage_{
 #else
     : __base_t(__base_t{typename __base_t::__stored_type{
 #endif
-        detail::__construct_psa_from_dynamic_values_tag_t<>{}, dyn
+      std::conditional_t<N==rank_dynamic(),
+        detail::__construct_psa_from_dynamic_values_tag_t<0>,
```

Comment:
Would you consider spelling out "psa" as we already do in `__construct_partially_static_array_from_{all,dynamic}_sizes_tag_t`?  That would improve consistency.

Reply:
I went the other way and now have:
```
__construct_psa_from_all_exts_values_tag_t
__construct_psa_from_dynamic_exts_values_tag_t
__construct_psa_from_all_exts_array_tag_t
__construct_psa_from_dynamic_exts_array_tag_t
```
For the four code paths.

Reply:
writing this out was just getting too long